Aquarius/Src/Aquarius/Core/Window.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+#include <string>
+#include <memory>
+#include <Aquarius/Renderer/RenderingContext.h>
+#include <GLFW/glfw3.h>
+
+namespace Aquarius {
+
+    // GLFW Window
+    class Window
+    {
+        using WindowPtr = std::unique_ptr<Window>;
+
+    public:
+        // Creation
+        Window(uint32_t width, uint32_t height, std::string&& name, bool vsync = false);
+        static WindowPtr Create(uint32_t width, uint32_t height, std::string&& name, bool vsync = false);
+
+        // Initialize GLFW
+        void Initialize();
+
+        // Poll inputs and swap buffers
+        void OnUpdate();
+
+        // Utility methods
+        uint32_t getWidth() const;
+        uint32_t getHeight() const;
+        const std::string& getName() const;
+        bool getVsync() const;
+        void setVsync(bool enabled);
+
+        // Destruction
+        ~Window();
```

Comment:
I like how you have ordered the methods almost "chronologically" in a way, makes it super easy to read

Reply:
This was intentional